File: backend/src/main/java/store/andefi/dto/OrderCheckoutResponseDto.java
package store.andefi.dto;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Data;

@Data
public class OrderCheckoutResponseDto {
  @JsonProperty("order_id")
  private String orderId;

  @JsonProperty("transaction_token")
  private String transactionToken;

  @JsonProperty("total_amount_before_discount")
  private long totalAmountBeforeDiscount;

  @JsonProperty("total_amount_after_discount")
  private long totalAmountAfterDiscount;

  @JsonProperty("total_discount")
  private long totalDiscount;
}

File: frontend/app/src/main/java/store/andefi/data/remote/dto/OrderCheckoutResponseDto.kt
package store.andefi.data.remote.dto

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class OrderCheckoutResponseDto(
    @SerialName("order_id") val orderId: String,
    @SerialName("transaction_token") val paymentRedirectUrl: String,
    @SerialName("total_amount_before_discount") val totalAmountBeforeDiscount: Long,
    @SerialName("total_amount_after_discount") val totalAmountAfterDiscount: Long,
    @SerialName("total_discount") val totalDiscount: Long,
)
